sdk/python/sawtooth_sdk/client/future.py

```diff
@@ -37,6 +44,9 @@ def result(self, timeout=None):
             if self._result is None:
                 if not self._condition.wait(timeout):
                     raise FutureTimeoutError('Future timed out')
+        if isinstance(self._result, FutureError):
+            raise ValidatorConnectionError(
```

Comment:
The error could be stored in the FutureError - keeping the error as part of the operation, not an interpretation of the error.